src/sagemaker/amazon/amazon_estimator.py

```diff
@@ -47,7 +49,7 @@ def __init__(self, role, train_instance_count, train_instance_type, data_locatio
         self.data_location = data_location
 
     def train_image(self):
-        return registry(self.sagemaker_session.boto_region_name) + "/" + type(self).repo
+        return registry(self.sagemaker_session.boto_region_name, type(self).__name__) + "/" + type(self).repo
```

Comment:
Good observation. Not sure about the scenario for subclassing these wrapper classes but in any case there are two options: either you add your new class to registry() mapping (just like you add any new class) or provide custom 'train_image' in a subclass.

Reply:
I'm talking about end users subclassing them in their own code. There's no super obvious case for doing so, but it's also not wrong to do so, and all else being equal there shouldn't be unexpected things that break when it happens.

The cost to make this work in the case of subclassing isn't big IMO - you can modify the registry() method to accept a Class instead of a string, then instead of just using direct equality checks, you use is https://docs.python.org/3/library/functions.html#issubclass instead.

Reply:
The implementation with class has an issue with circular dependency, let's use same approach as with repo name then